lib/phoenix_live_view/upload_config.ex

```diff
@@ -0,0 +1,570 @@
+
+    recalculate_computed_fields(%UploadConfig{conf | entries: [], errors: []})
+  end
+
+  defp cast_and_validate_entry(%UploadConfig{} = conf, %{"ref" => ref} = client_entry) do
+    entry = %UploadEntry{
+      ref: ref,
+      upload_ref: conf.ref,
+      upload_config: conf.name,
+      client_name: Map.fetch!(client_entry, "name"),
+      client_size: Map.fetch!(client_entry, "size"),
+      client_type: Map.fetch!(client_entry, "type"),
+      client_last_modified: Map.get(client_entry, "last_modified")
+    }
+
+    {:ok, entry}
+    |> validate_max_file_size(conf)
+    |> validate_accepted(conf)
+    |> case do
+      {:ok, entry} ->
+        {:ok, put_valid_entry(conf, entry)}
+
+      {:error, reason} ->
+        {:error, put_invalid_entry(conf, entry, reason)}
+    end
+  end
+
+  defp put_valid_entry(conf, entry) do
+    entry = %UploadEntry{entry | valid?: true}
+    new_pids = Map.put(conf.entry_refs_to_pids, entry.ref, @unregistered)
+    new_metas = Map.put(conf.entry_refs_to_metas, entry.ref, %{})
+
+    %UploadConfig{
+      conf
+      | entries: conf.entries ++ [entry],
+        entry_refs_to_pids: new_pids,
+        entry_refs_to_metas: new_metas
+    }
+  end
+
+  defp put_invalid_entry(conf, entry, reason) do
+    entry = %UploadEntry{entry | valid?: false}
+    new_pids = Map.put(conf.entry_refs_to_pids, entry.ref, @invalid)
+    new_metas = Map.put(conf.entry_refs_to_metas, entry.ref, %{})
+
+    new_conf = %UploadConfig{
+      conf
+      | entries: conf.entries ++ [entry],
+        entry_refs_to_pids: new_pids,
+        entry_refs_to_metas: new_metas
+    }
+
+    put_error(new_conf, entry.ref, reason)
+  end
+
+  defp validate_max_file_size({:ok, %UploadEntry{client_size: size}}, %UploadConfig{
+         max_file_size: max
+       })
+       when size > max or not is_integer(size),
+       do: {:error, :too_large}
+
+  defp validate_max_file_size({:ok, entry}, _conf), do: {:ok, entry}
+
+  defp validate_accepted({:ok, %UploadEntry{} = entry}, conf) do
+    if accepted?(conf, entry) do
+      {:ok, entry}
+    else
+      {:error, :not_accepted}
+    end
+  end
+
+  defp validate_accepted({:error, _} = error, _conf), do: error
+
+  defp accepted?(%UploadConfig{accept: :any}, _entry), do: true
+
+  defp accepted?(%UploadConfig{accept: %{"image/*" => _}}, %UploadEntry{
+         client_type: <<"image/" <> _>>
+       }),
+       do: true
+
+  defp accepted?(%UploadConfig{accept: %{"audio/*" => _}}, %UploadEntry{
+         client_type: <<"audio/" <> _>>
+       }),
+       do: true
+
+  defp accepted?(%UploadConfig{accept: %{"video/*" => _}}, %UploadEntry{
+         client_type: <<"video/" <> _>>
+       }),
+       do: true
+
+  defp accepted?(%UploadConfig{accept: accept}, %UploadEntry{} = entry) do
+    cond do
+      Map.has_key?(accept, entry.client_type) -> true
+      Path.extname(entry.client_name) in (accept |> Map.values() |> Enum.concat()) -> true
```

Comment:
You never access the values alongside the key, so it feels this should be two separate internal states: "accepted_types" and "accepted_exts". Or make them a `accepted: {types, exts}` tuple.